airflow/models/dag.py

```diff
@@ -1394,36 +1402,54 @@ def sub_dag(self, *args, **kwargs):
         return self.partial_subset(*args, **kwargs)
 
     def partial_subset(
-        self, task_regex, include_downstream=False, include_upstream=True
+        self,
+        task_ids_or_regex: Union[str, PatternType, Iterable[str]],
+        include_downstream=False,
+        include_upstream=True,
+        include_direct_parents=False,
     ):
         """
         Returns a subset of the current dag as a deep copy of the current dag
         based on a regex that should match one or many tasks, and includes
         upstream and downstream neighbours based on the flag passed.
+
+        :param task_ids_or_regex: Either a list of task_ids, or a regex to
+            match against task ids (as a string, or compiled regex pattern).
+        :type task_ids_or_regex: [str] or str or re.Pattern
+        :param include_downstream: Include all downstream tasks of matched
+            tasks, in addition to matched tasks.
+        :param include_upstream: Include all upstream tasks of matched tasks,
+            in addition to matched tasks.
         """
         # deep-copying self.task_dict and self._task_group takes a long time, and we don't want all
         # the tasks anyway, so we copy the tasks manually later
         task_dict = self.task_dict
         task_group = self._task_group
         self.task_dict = {}
-        self._task_group = None
+        self._task_group = None  # type: ignore
         dag = copy.deepcopy(self)
         self.task_dict = task_dict
         self._task_group = task_group
 
-        regex_match = [
-            t for t in self.tasks if re.findall(task_regex, t.task_id)]
+        if isinstance(task_ids_or_regex, (str, PatternType)):
+            matched_tasks = [
+                t for t in self.tasks if re.findall(task_ids_or_regex, t.task_id)]
+        else:
+            matched_tasks = [t for t in self.tasks if t.task_id in task_ids_or_regex]
+
         also_include = []
-        for t in regex_match:
+        for t in matched_tasks:
             if include_downstream:
                 also_include += t.get_flat_relatives(upstream=False)
             if include_upstream:
                 also_include += t.get_flat_relatives(upstream=True)
+            if include_direct_parents and not include_upstream:
+                also_include += t.get_flat_relatives(upstream=True, recurse=False)
```

Comment:
```suggestion
            elif include_direct_parents:
                also_include += t.upstream_list
```

And then we don't need to add `recurse=True` parameter to get_flat_relatives (the "flat" part is to do the flattening/recursing part)